fix(linter): `unicorn/filename-case` report only on the first js block for `astro`, `vue` and `svelte` files

Sysix opened the pull request:
before:
reported for each `<script>` block, the diagnostic
after:
only for the first `<script>` block.

camc314 commented:
### Merge activity

* **Oct 15, 9:03 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 15, 9:03 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
fix(linter): `unicorn/filename-case` report only on the first js block for `astro`, `vue` and `svelte` files (#14599)

before:
reported for each `<script>` block, the diagnostic
after:
only for the first `<script>` block.
